awscli/customizations/s3/filegenerator.py

```diff
@@ -304,10 +302,8 @@ def _list_single_object(self, s3_path):
         # IAM policies with the smallest set of permissions needed) and
         # instead use a HeadObject request.
         bucket, key = find_bucket_key(s3_path)
-        operation = self._service.get_operation('HeadObject')
         try:
-            response = operation.call(
-                self._endpoint, bucket=bucket, key=key)[1]
+            response = self._client.head_object(Bucket=bucket, Key=key)
         except ClientError as e:
             # We want to try to give a more helpful error message.
             # This is what the customer is going to see so we want to
```

Comment:
Here is the error message using botocore clients:
```
In [1]: import botocore.session

In [2]: session = botocore.session.get_session()

In [3]: s3 = session.create_client('s3')

In [4]: s3.head_object(Bucket='mybucketfoo', Key='some_key')
---------------------------------------------------------------------------
ClientError                               Traceback (most recent call last)
<ipython-input-4-2bede845a716> in <module>()
----> 1 s3.head_object(Bucket='mybucketfoo', Key='some_key')

/Users/kyleknap/Documents/GitHub/botocore/botocore/client.pyc in _api_call(self, **kwargs)
    324 
    325             if http.status_code >= 300:
--> 326                 raise ClientError(parsed_response, operation_name)
    327             else:
    328                 return parsed_response

ClientError: An error occurred (404) when calling the HeadObject operation: Not Found
```
Here is the error from s3:
```
A client error (NoSuchKey) occurred when calling the HeadObject operation: Key "some_key" does not exist
```
I like the second one better. You could figure out that the object does not exist using the first exception message, but it is not very explicit. @jamesls What is your opinion?